guppylang/definition/ty.py

```diff
@@ -1,20 +1,24 @@
 from abc import abstractmethod
 from collections.abc import Callable, Sequence
 from dataclasses import dataclass, field
-from typing import TYPE_CHECKING
+from typing import TYPE_CHECKING, TypeAlias
 
 from hugr.serialization import tys
 
 from guppylang.ast_util import AstNode
 from guppylang.definition.common import CompiledDef, Definition
+from guppylang.error import GuppyError
 from guppylang.tys.arg import Argument
 from guppylang.tys.param import Parameter, check_all_args
-from guppylang.tys.ty import OpaqueType, Type
+from guppylang.tys.ty import InputFlags, OpaqueType, Type
 
 if TYPE_CHECKING:
     from guppylang.checker.core import Globals
 
 
+FlaggedArgs: TypeAlias = Sequence[tuple[Argument, InputFlags]]
```

Comment:
Most uses have gone from `Sequence[Argument]` to `FlaggedArgs`, but you do have a couple of `tuple[Argument, InputFlags]` elsewhere. Maybe define `FlaggedArg = tuple[...]` and use that (so `Sequence[FlaggedArg]` in many places)....albeit, elsewhere I suggest a bigger redifinition of FlaggedArg ;)

Reply:
This no longer applied after ff545a7